python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -273,95 +303,208 @@ def _clean_exclude(exclude, feature_map):
     return ret
 
 
+@dataclass(repr=False)
 class EBMModel(BaseEstimator):
-    """Base class for all EBMs"""
+    """Base class for all EBMs."""
+
+    # Explainer
+    feature_names: Optional[Sequence[str]] = None
+    feature_types: Optional[Sequence[Optional[str]]] = None
+    # Preprocessor
+    max_bins: NonNegativeInt = 1024
+    max_interaction_bins: NonNegativeInt = 32
+    # Stages
+    interactions: Optional[Union[
+        Annotated[int, Field(strict=True, ge=0)],
+        Annotated[float, Field(gt=0.0, lt=1.0)],
+        List[Union[tuple, Sequence[Union[int, str]]]],
+    ]] = 0.95
+    exclude: Optional[Union[
+        Literal["mains"],
+        List[Union[Sequence[Union[int, str]], str, int]],
+    ]] = None
+    # Ensemble
+    validation_size: Union[
+        Annotated[int, Field(strict=True, ge=0)],
+        Annotated[float, Field(ge=0.0, lt=1.0)],
+    ] = 0.15
+    outer_bags: Annotated[int, Field(ge=1)] = 14
+    inner_bags: NonNegativeInt = 0
+    # Boosting
+    learning_rate: PositiveFloat = 0.01
+    greediness: NonNegativeFloat = 1.5
+    cyclic_progress: Annotated[float, Field(ge=0.0, le=1.0)] = 1.0
+    smoothing_rounds: NonNegativeInt = 200
+    interaction_smoothing_rounds: NonNegativeInt = 50
```

Comment:
The following attributes do not make sense on the DPEBM class and should only exist on the regular EBM class: max_interaction_bins, interactions, inner_bags, greediness, cyclic_progress, smoothing_rounds, interaction_smoothing_rounds, early_stopping_rounds, early_stopping_tolerance, min_samples_leaf, min_hessian

I think to do this we'll need to have separate a separate non-DP EBM class (EBMModel) and the base class for all the attributes which are shared between the DP and non-DP classes (EBMBase).